src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -527,7 +530,7 @@ private void WriteResponseHeadersUnsynchronized(int streamId, int statusCode, Ht
         catch (Exception ex)
         {
             _log.HPackEncodingError(_connectionId, streamId, ex);
-            _http2Connection.Abort(new ConnectionAbortedException(ex.Message, ex));
+            _http2Connection.Abort(new ConnectionAbortedException(ex.Message, ex), Http2ErrorCode.INTERNAL_ERROR, ConnectionEndReason.ErrorWritingHeaders);
```

Comment:
Do you mean too broad? The most common error is probably encoding a header to hpack because it contains invalid characters. At this level we don't have good insight to the different errors.

`ErrorWritingHeaders` and `ErrorReadingHeaders` are two high level reasons for bad errors. I'm not sure how to break them down better.

Reply:
No, (I believe) I meant "can this be lumped in with another end reason?".  Is it useful to know that it was the headers specifically that failed or is it sufficient to know that the response failed?